src/app/component/ProductCard/ProductCard.component.js

```diff
@@ -101,12 +116,30 @@ export class ProductCard extends PureComponent {
     }
 
     renderProductPrice() {
-        const { product: { price_range } } = this.props;
+        const { product: { price_range, type_id } } = this.props;
 
         if (!price_range) {
             return <TextPlaceholder />;
         }
 
+        if (type_id === CONFIGURABLE) {
+            const { product: { variants } } = this.props;
+            if (this.isConfigurableProductOutOfStock(variants)) {
+                return null;
+            }
+        }
+        if (type_id === BUNDLE) {
+            const { product: { items } } = this.props;
+
+            if (items.length === 0) {
+                return null;
+            }
+
+            if (this.isBundleProductOutOfStock(items[0])) {
+                return null;
+            }
```

Comment:
The same thing here, if you have a calculation here, why not move `items.length === 0` and the `const { product: { items } } = this.props;` retrieval to this function?